node/connection_base.js

```diff
@@ -71,6 +73,31 @@ function TChannelConnectionBase(channel, direction, remoteAddr) {
 
     self.tracer = self.channel.tracer;
     self.ops.startTimeoutTimer();
+
+    if (direction === 'out') {
+        self.channel.connectionsInitiatedStat.increment(1, {
+            'host-port': self.channel.hostPort || '0.0.0.0:0',
+            'peer-host-port': self.remoteAddr
+        });
+    } else {
+        self.channel.connectionsAcceptedStat.increment(1, {
+            'host-port': self.channel.hostPort,
+            'peer-host-port': self.remoteAddr
+        });
+    }
+
+    if (self.reportInterval > 0) {
+        self.reportTimer = self.timers.setTimeout(
+            onReport, self.reportInterval
+        );
+    }
+
+    function onReport() {
+        self.channel.connectionsActiveStat.update(1, {
```

Comment:
That's not how gauges work.

You need to `update(totalNum)`.

The only person that has access to that is the rootChannel. It tracks all the incoming connections in serverConnections and it has access to all the peers so in theory can track outgoing connections.